x-pack/plugins/apm/public/components/app/service_overview/service_overview_errors_table/get_columns.tsx

```diff
@@ -28,12 +37,37 @@ export function getColumns({
   serviceName,
   errorGroupDetailedStatistics,
   comparisonEnabled,
+  urlParams,
 }: {
   serviceName: string;
   errorGroupDetailedStatistics: ErrorGroupDetailedStatistics;
   comparisonEnabled?: boolean;
+  urlParams: ApmUrlParams;
 }): Array<EuiBasicTableColumn<ErrorGroupMainStatistics['errorGroups'][0]>> {
   return [
+    {
+      name: i18n.translate('xpack.apm.errorsTable.typeColumnLabel', {
+        defaultMessage: 'Type',
+      }),
+      field: 'type',
+      sortable: false,
+      render: (_, { type }) => {
+        return (
+          <ErrorLink
```

Comment:
We should no longer use the link helpers in `apm/public/components/shared/links/` that rely on `useAPMHref` and `getLegacyApmHref`.

Instead we should use the [new link client](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/dev_docs/routing_and_linking.md).

I renamed `getApmHref` to `getLegacyApmHref` to make this more obvious but I can see that it's still not very clear. 

Can you try making the change to the new link client, and then think of ways to make it more apparent what are the legacy methods, and what to use instead?

Eg perhaps rename `useAPMHref` to `useLegacyApmHref` and add a code comment with a link to https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/dev_docs/routing_and_linking.md

Reply:
If this change becomes too big it's also fine to postpone it for a follow-up PR